lua/cmp_nvim_ultisnips/config.lua

```diff
@@ -5,6 +5,7 @@ local M = {}
 M.default_config = {
   show_snippets = "expandable",
   documentation = cmpu_snippets.documentation,
+  filetype_source = "filename", -- TODO: better name for filename?
```

Comment:
Any ideas for a better name?

Reply:
we can mention `filetype_source` from the README, something like: determine how filetype is indicated to load the snippets ... 

I think we could remove the option `filename`, I would suggest to set `filetype_source` to `treesitter` if treesitter is installed. Because of the followings:
 - Vim prefers to detect ft by treesitter if it's installed, we should do it as well
 - Not to make the config step of our plugin more complicated
 
How do you think?

Reply:
These are great suggestions! I am currently unsure if we should either do:
1. add options `auto`, `treesitter`, `filetype` (default `auto`)
2. add options `treesitter`, `filetype` (default `treesitter`)

For 1. if set to `treesitter`, the user will be warned if treesitter is not found instead of falling back to the default ft detection silently.

However, I kind of like the simplicity of your suggestions and agree that we should not overcomplicate things for users. So I prefer 2. at the moment.

Reply:
These are great suggestions! I am currently unsure if we should include the option `both`
so that e.g. in a \`\`\` block in markdown you will get either:
1. only snippets for the language of that block
2. also snippets for the filetype of the buffer

I don't know if this so useful that we should include it. Please let me know what you think!